{
  try {
    return future.get();
  }
 catch (  InterruptedException ie) {
    throw ie;
  }
catch (  Exception e) {
    return attemptConvert(e);
  }
}
